test/test_ops.py

```diff
@@ -1113,277 +1111,48 @@ def test_bbox_convert_jit(self):
-
-    assert_distance_iou_loss(box1, box3, 1.1923)
-
-    assert_distance_iou_loss(box1, box4, 1.2500)
-
-    assert_distance_iou_loss(box1s, box2s, 1.2250, reduction="mean")
-    assert_distance_iou_loss(box1s, box2s, 2.4500, reduction="sum")
-
-
-@pytest.mark.parametrize("device", cpu_and_gpu())
-@pytest.mark.parametrize("dtype", [torch.float32, torch.half])
-def test_empty_distance_iou_inputs(dtype, device) -> None:
-    box1 = torch.randn([0, 4], dtype=dtype, device=device).requires_grad_()
-    box2 = torch.randn([0, 4], dtype=dtype, device=device).requires_grad_()
-
-    loss = ops.distance_box_iou_loss(box1, box2, reduction="mean")
-    loss.backward()
-
-    tol = 1e-3 if dtype is torch.half else 1e-5
-    torch.testing.assert_close(loss, torch.tensor(0.0, device=device), rtol=tol, atol=tol)
-    assert box1.grad is not None, "box1.grad should not be None after backward is called"
-    assert box2.grad is not None, "box2.grad should not be None after backward is called"
-
-    loss = ops.distance_box_iou_loss(box1, box2, reduction="none")
-    assert loss.numel() == 0, "diou_loss for two empty box should be empty"
-
-
-class TestCompleteBoxIou(BoxTestBase):
-    def _target_fn(self) -> Tuple[bool, Callable]:
-        return (True, ops.complete_box_iou)
-
-    def _generate_int_input() -> List[List[int]]:
-        return [[0, 0, 100, 100], [0, 0, 50, 50], [200, 200, 300, 300]]
-
-    def _generate_int_expected() -> List[List[float]]:
-        return [[1.0, 0.25, 0.0], [0.25, 1.0, 0.0], [0.0, 0.0, 1.0]]
-
-    def _generate_float_input() -> List[List[float]]:
-        return [
-            [285.3538, 185.5758, 1193.5110, 851.4551],
-            [285.1472, 188.7374, 1192.4984, 851.0669],
-            [279.2440, 197.9812, 1189.4746, 849.2019],
-        ]
-
-    def _generate_float_expected() -> List[List[float]]:
-        return [[1.0, 0.9933, 0.9673], [0.9933, 1.0, 0.9737], [0.9673, 0.9737, 1.0]]
-
-    @pytest.mark.parametrize(
-        "test_input, dtypes, tolerance, expected",
-        [
-            pytest.param(
-                _generate_int_input(), [torch.int16, torch.int32, torch.int64], 1e-4, _generate_int_expected()
-            ),
-            pytest.param(_generate_float_input(), [torch.float32, torch.float64], 0.002, _generate_float_expected()),
-            pytest.param(_generate_float_input(), [torch.float32, torch.float64], 0.001, _generate_float_expected()),
-        ],
-    )
-    def test_complete_iou(self, test_input: List, dtypes: List[torch.dtype], tolerance: float, expected: List) -> None:
-        self._run_test(test_input, dtypes, tolerance, expected)
+            dtype=dtype,
+        )
+        expected = torch.tensor([604723.0806, 600965.4666, 592761.0085], dtype=torch.float64)
+        area_check(box_tensor, expected, tolerance=0.05)
 
-    def test_ciou_jit(self) -> None:
-        self._run_jit_test([[0, 0, 100, 100], [0, 0, 50, 50], [200, 200, 300, 300]])
+    def test_float16_box(self):
```

Comment:
We can probably make stuff a bit compact with parametrize with pytest.params. Hmm but I was like it's ok this isn't so bad?

Reply:
Do you know why this check is special cases and not grouped with the other float dtypes?

Reply:
Not sure probably to check overflow from float16 to float32?

Reply:
Could you please check if the test above works with float16 and potentially remove this one?